src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer.Protocol/DelegatedTypes.cs

```diff
@@ -57,3 +64,10 @@ internal record DelegatedCompletionItemResolveParams(
     VersionedTextDocumentIdentifier HostDocument,
     VSInternalCompletionItem CompletionItem,
     RazorLanguageKind OriginatingKind);
+
+internal record DelegatedProjectContextsParams(
+    VersionedTextDocumentIdentifier HostDocument);
+
+internal record DelegatedDocumentSymbolParams(
+    VersionedTextDocumentIdentifier HostDocument,
+    VSProjectContext? ProjectContext);
```

Comment:
Why send the project context separately, rather than as part of the test identifier, like the rest of LSP does? I would have thought we could do this whole thing without changing any of these DTOs, just make sure the project context was carried along when we create new document identifiers?

Reply:
Okay the folding range DTO looks like its "special" and would need the change, but maybe that could be brought in line with the rest?

Also, why does completion, onautoinsert, etc. not need this?

Reply:
I found the full list of impacted endpoints. Added to the PR description and will checkbox as I audit each of them

Reply:
> I found the full list of impacted endpoints

Is it not just all of the endpoints that have a TextDocumentIdentifier? Thats why I think if we can just carry the context along with that, it will be easier, and more future proof.

Reply:
Sort of. There are places where we don't have the context yet. That said, I think one thing we can do is change `IRazorRequestHandler` to return `VSTextDocumentIdentifier` as the type instead of `TextDocumentIdentifier`. Then provide null for cases it's not known, which will also point to areas we need to decide what TFM support we're going to add and how.

Reply:
![image](https://github.com/dotnet/razor/assets/475144/e35b6b81-3938-4e79-ab55-16747418488e)
![ByeFeliciaHidingGIF](https://github.com/dotnet/razor/assets/475144/9c491b3a-75de-4e55-8e5d-84c1499e91a8)

Reply:
On my phone, so haven't looked at the code change, but that's very good in theory 👍